### PR TITLE
feat: site with mdbook

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

TODO

- [x] set up mdbook with same file structure (i.e. /cips stays, becomes default src dir, and /wgs move to /cips/wgs)
- [x] custom handling for frontmatter in CIPs, use option 2 from here https://github.com/celestiaorg/CIPs/pull/170#issuecomment-2239814656
- [x]  set up dns
- [x]  test out on personal github pages https://cips.jcstein.dev
- [x]  prep for plausible
    - [x]  add plausible script
- [x]  favicon
- [x]  prep site for dns

The previous PR got too messy and I'd like to try without moving the structure around. https://github.com/celestiaorg/CIPs/pull/168

### CODE REVIEW

**Comment by jcstein** [comment]
https://jcstein.github.io/CIPs/index.html

**Comment by jcstein** [comment]
Options for frontmatter:

# 1. change the styling to have backticks around it and display the code block
Screenshot 2024-07-19 at 1 54 22 PM

this also works on github
Screenshot 2024-07-19 at 1 59 10 PM

# 2. change the style of the frontmatter itself to be .md-friendly instead of github-flavored-markdown friendly. IMHO this looks worse
Screenshot 2024-07-19 at 2 00 29 PM

**Comment by jcstein** [comment]
edge case for option 3 is that you can't see the whole table
Screenshot 2024-07-19 at 2 08 31 PM

but if it is a code block, you can
Screenshot 2024-07-19 at 2 09 14 PM

**Comment by jcstein** [comment]
i'm actually not sure what's best here 😅 
it's pretty janky to scroll through the code block
Screenshot 2024-07-19 at 2 49 11 PM

**Comment by jcstein** [comment]
**going with option 3**

**Comment by jcstein** [comment]
**demo: https://cips.jcstein.dev/**
diff for final touches: https://github.com/celestiaorg/CIPs/compare/jcs/add-site-v0...jcstein:CIPs:main

once this is approved, 2 things will need to happen before going live:
1. @sysrex to add DNS for cips.celestia.org for github pages deployment. this should just be adding a CNAME entry for "cips" since gh pages is already setup for the A records of celestia.org
2. @musalbas to begin collecting plausible analytics. it's already set up on the site's end https://github.com/celestiaorg/CIPs/blob/2bedecfa0fd9d0f11145859c0d633eeb429c9000/theme/head.hbs#L1

File: .github/workflows/mdbook.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Build job
+  build:
+    runs-on: ubuntu-latest
+    env:
+      MDBOOK_VERSION: 0.4.36
```
Comment (rootulp) [inline .github/workflows/mdbook.yml:28]: [optional] looks like 0.4.40 is latest: https://github.com/rust-lang/mdBook/releases/tag/v0.4.40

```suggestion
      MDBOOK_VERSION: 0.4.40
```

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,8 @@
+book
+index.html
+
+# CIPs
+cips/*.html
+
+# WGs
+cips/wgs/*/*.html
```
Comment (rootulp) [inline .gitignore:8]: should this have two asterisks to match any nested subdirectories?

```suggestion
cips/wgs/**/*.html
```

File: .markdownlint.yaml [hunk 3]
```diff
@@ -3,3 +3,6 @@
   "code_blocks": false # Disable rule for hard tabs in code blocks
 "MD013": false # Disable rule for line length
 "MD033": false # Disable rule banning inline HTML
+"MD025": false # Disable rule for multiple top-level headers
```
Comment (rootulp) [inline .markdownlint.yaml:6]: [no change needed] I like this rule. Are we violating a lot (hence why we ignore it here)?

File: README.md [hunk 4]
```diff
@@ -54,3 +54,11 @@ Files in this repo must conform to [markdownlint](https://github.com/DavidAnson/
...
 markdownlint --config .markdownlint.yaml '**/*.md'
 ```
+
+### Running the site locally
+
+```bash
+mdbook serve -o
+```
+
+> Requires Rust and mdbook.
```
Comment (rootulp) [inline README.md:64]:
```suggestion
### Running the site locally

Prerequisites: 
1. Install [Rust](https://www.rust-lang.org/tools/install)
1. Install [mdbook](https://rust-lang.github.io/mdBook/guide/installation.html)

```bash
mdbook serve -o
```
```

File: cips/cip-1.md [hunk 5]
```diff
@@ -316,6 +311,8 @@ format. There is a [CIP template](./cip-template.md) to follow.
 Each CIP must begin with an RFC 822 style header preamble, preceded
 and followed by three hyphens (---). This header is also termed
 “front matter” by Jekyll. The headers must appear in the following order.
+In order to display on the CIP site, the frontmatter must also
+be surrounded by three backticks (```).
```
Comment (rootulp) [inline cips/cip-1.md:315]: [nit] I think the approach taken in this PR is to use a markdown table so perhaps this needs to be updated.

```suggestion
In order to display on the CIP site, the frontmatter must be
formatted in a markdown table.
```

**Review by rootulp — APPROVED** [review]
Overall LGTM, thanks for all your hard work on this!

**Comment by musalbas** [comment]
GFM style look very annoying to edit for CIP authors and look worse rendered to me. Why do you think it looks worse than option 2?

File: .markdownlint.yaml (hunk 3 above)
Comment (jcstein) [reply .markdownlint.yaml:6]: good callout, i'll add it just to the SUMMARY.md file for the sidebar structure

**Comment by jcstein** [comment]
I think options 2 and 3 are equally as annoying to edit/author and neither is perfect. GFM just fits into the current style that is displayed on github

**Comment by musalbas** [comment]
I don't follow how, as option 2 just requires the author to write bulletpoints whereas option 2 requires them to add a bunch of ------ ascii art and scroll right to see the whole text

**Comment by jcstein** [comment]
I see, the table for option 2 is easier to edit. i will demo on this PR

```
| cip | 1 |
| - | - |
| title | Celestia Improvement Proposal Process and Guidelines |
| author | Yaz Khoury <yaz@celestia.org> |
| status | Living |
| type | Meta |
| created | 2023-04-13 |
```

renders:
| cip | 1 |
| - | - |
| title | Celestia Improvement Proposal Process and Guidelines |
| author | Yaz Khoury <yaz@celestia.org> |
| status | Living |
| type | Meta |
| created | 2023-04-13 |

File: .markdownlint.yaml (hunk 3 above)
Comment (jcstein) [reply .markdownlint.yaml]: https://github.com/celestiaorg/CIPs/pull/170/commits/b093bb02136fa161eec9a6f7b6f447396e149840

File: .github/workflows/mdbook.yml (hunk 1 above)
Comment (jcstein) [reply .github/workflows/mdbook.yml]: accidentally committed this instead of using your suggestion, myb !

File: README.md (hunk 4 above)
Comment (jcstein) [reply README.md]: done! https://github.com/celestiaorg/CIPs/pull/170/commits/4d490dc2acce7928644d925992933b4e3350cf3a ty

**Pull request merged**